contracts/allocators/OnsenAllocatorV2.sol

```diff
@@ -0,0 +1,268 @@
+        return rewards;
+    }
+
+    function utilityTokens() public view override returns (IERC20[] memory) {
+        IERC20[] memory utility = new IERC20[](1);
+        utility[0] = IERC20(sushi);
+        return utility;
+    }
+
+    function name() external pure override returns (string memory) {
+        return "OnsenAllocator";
+    }
+
+    function setTreasury(address treasuryAddress) external onlyGuardian {
+        treasury = treasuryAddress;
+    }
+
+    function setSushi(address sushiAddress) external onlyGuardian {
+        sushi = sushiAddress;
+    }
+
+    function setXSushi(address xSushiAddress) external onlyGuardian {
+        xSushi = xSushiAddress;
+    }
+
+    function setMasterChefAddress(address masterChefAddress) external onlyGuardian {
+        masterChef = masterChefAddress;
+    }
+
+    /// ========== INTERNAL FUNCTIONS ==========
+
+    /**
+     * @notice stake sushi rewards
+     */
+    function _stakeSushi() internal {
+        uint256 balance = IERC20(sushi).balanceOf(address(this));
+        if (balance > 0) {
+            IERC20(sushi).approve(xSushi, balance);
+            ISushiBar(xSushi).enter(balance); // stake sushi
+        }
+    }
+
+    /**
+     * @notice unStake sushi rewards
+     */
+    function _unStakeSushi() internal {
+        uint256 balance = IERC20(xSushi).balanceOf(address(this));
+        if (balance > 0) {
+            ISushiBar(xSushi).leave(balance); // unstake $xSUSHI
+        }
+    }
+
+    /**
+     *  @notice pending $SUSHI rewards
+     *  @return uint
+     */
+    function _pendingSushi(uint256 pid) internal view returns (uint256) {
+        return IMasterChef(masterChef).pendingSushi(pid, address(this));
+    }
+
+    /**
+     * @notice Find out pool Id from Onsen based on the LP token
+     * according to Onsen documentation there shouldn't be more than pool by LPToken, so we are going to use the first ocurrence of the LPToken in the Onsen pools
+     * @return [bool, uint256], first value shows if value was found, second value the id for the pool
+     */
+    function _findPoolByLP(address LPToken) internal view returns (bool, uint256) {
+        /// Check if the id is already stored in the local variable
+        PoolData memory pd = _lpToOnsenId[LPToken];
+
+        /// If the pool is active return the id else search for the pool id
+        if (pd.poolActive) {
+            return (pd.poolActive, pd.id);
+        } else {
+            /// Call the poolLength function from sushi masterchef v2 and compare against the LP token passed as parameter
+            uint256 poolsLength = IMasterChef(masterChef).poolLength();
+
+            for (uint256 i; i < poolsLength; i++) {
+                /// If found, return the the pool Id from Onsen for the current LPToken.
+                if (LPToken == (address)(IMasterChef(masterChef).lpToken(i))) {
+                    pd = _lpToOnsenId[LPToken];
+                    pd.poolActive = true;
```

Comment:
if were changing variables here this not a view function anymore right though the compiler would say something.

Reply:
![image](https://user-images.githubusercontent.com/3720650/160556051-21dc7d3f-da2f-4b9d-933b-4d694e4b72f0.png)

I wasn't sure about this one since Visual Studio is suggesting me to make it "view", because it is finding that the only variable that is changing was declared inside the function, but it is actually changing a state variable so yes, I just change it.